diff -Nur vtun/Makefile.in vtun.linking/Makefile.in
--- vtun/Makefile.in	2002-12-20 17:55:47.000000000 +0100
+++ vtun.linking/Makefile.in	2006-08-03 04:29:13.000000000 +0200
@@ -19,7 +19,8 @@
 #  
 CC = @CC@
 CFLAGS = @CFLAGS@ @CPPFLAGS@
-LDFLAGS = @LDFLAGS@ @LIBS@
+LIBS = @LIBS@
+LDFLAGS = @LDFLAGS@
 
 YACC = @YACC@
 YACCFLAGS = -d
@@ -58,7 +59,7 @@
 	$(CC) $(CFLAGS) $(DEFS) -c $<
 
 vtund: $(OBJS)
-	$(CC) $(CFLAGS) -o vtund $(OBJS) $(LFD_OBJS) $(LDFLAGS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -o vtund $(OBJS) $(LFD_OBJS) $(LIBS)
 
 cfg_file.tab.c: cfg_file.y cfg_kwords.h config.h
 	$(YACC) $(YACCFLAGS) -b cfg_file cfg_file.y
