--- vtun/configure.in.orig	Fri Dec 20 18:13:03 2002
+++ vtun/configure.in	Mon Feb 23 17:13:49 2004
@@ -134,7 +134,7 @@
    AC_MSG_RESULT()
    AC_CHECKING( for md5 Library and Header files ... )
 AC_SEARCH_HEADERS(md5.h,
-    $SSL_HDR_DIR /usr/include/openssl "" /usr/include /usr/include/ssl /usr/local/include /usr/local/ssl/include, /usr/include/sys, 
+    $SSL_HDR_DIR /usr/include/openssl "" /usr/include /usr/include/ssl /usr/local/include /usr/local/ssl/include /usr/include/sys, 
     ,
     AC_MSG_ERROR( SSL headers not found. )
    )
