diff -Nur vtun_/configure.in vtun/configure.in
--- vtun_/configure.in	2005-07-18 22:54:46.064184856 +0200
+++ vtun/configure.in	2005-07-18 22:55:07.060992856 +0200
@@ -119,9 +119,9 @@
    AC_CHECKING( for LZO Library and Header files ... )
    AC_SEARCH_HEADERS(lzo1x.h, 
 	$LZO_HDR_DIR /usr/include/lzo "" /usr/local/include,
-        AC_CHECK_LIB(lzo, lzo1x_decompress,
+        AC_CHECK_LIB(lzo2, lzo1x_decompress,
 	    [
-	       LIBS="$LIBS -llzo"
+	       LIBS="$LIBS -llzo2"
 	       AC_DEFINE(HAVE_LZO)
 	    ],
 	    AC_MSG_ERROR( LZO library not found. )
diff -Nur vtun_/lfd_lzo.c vtun/lfd_lzo.c
--- vtun_/lfd_lzo.c	2002-04-25 11:19:50.000000000 +0200
+++ vtun/lfd_lzo.c	2005-07-18 22:51:20.000000000 +0200
@@ -36,6 +36,7 @@
 #ifdef HAVE_LZO
 
 #include "lzo1x.h"
+#include "lzoutil.h"
 
 static lzo_byte *zbuf;
 static lzo_voidp wmem;
