diff -ruN vtun-2.3.orig/Makefile.in vtun-2.3/Makefile.in
--- vtun-2.3.orig/Makefile.in	Mon May 29 15:35:11 2000
+++ vtun-2.3/Makefile.in	Sat Sep 23 14:17:28 2000
@@ -27,8 +27,9 @@
 LEX = @LEX@
 LEXFLAGS = -t 
 
+DESTDIR =
+
 INSTALL = @INSTALL@
-INSTALL_OWNER = -o root -g 0
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
@@ -79,24 +80,20 @@
 	rm -f `cat vtun.drivers`	
 
 install_man: 
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(MAN_DIR)/man8
-	$(INSTALL) -m 644 $(INSTALL_OWNER) vtund.8 $(MAN_DIR)/man8
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(MAN_DIR)/man5
-	$(INSTALL) -m 644 $(INSTALL_OWNER) vtund.conf.5 $(MAN_DIR)/man5
-	rm -f $(MAN_DIR)/man8/vtun.8 
-	ln -s vtund.8 $(MAN_DIR)/man8/vtun.8 
+	$(INSTALL) -d -m 755 $(DESTDIR)$(MAN_DIR)/man8
+	$(INSTALL) -m 644 vtund.8 $(DESTDIR)$(MAN_DIR)/man8
+	$(INSTALL) -d -m 755 $(DESTDIR)$(MAN_DIR)/man5
+	$(INSTALL) -m 644 vtund.conf.5 $(DESTDIR)$(MAN_DIR)/man5
 
 install_config: 
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(ETC_DIR)
-	if [ ! -f $(ETC_DIR)/vtund.conf ]; then \
-	  $(INSTALL) -m 600 $(INSTALL_OWNER) vtund.conf $(ETC_DIR); \
-	fi
+	$(INSTALL) -d -m 755 $(DESTDIR)$(ETC_DIR)
+	$(INSTALL) -m 600 vtund.conf $(DESTDIR)$(ETC_DIR);
 
 install: vtund install_config install_man
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(VAR_DIR)/run
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(STAT_DIR)
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(LOCK_DIR)
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(SBIN_DIR)
-	$(INSTALL) -m 755 $(INSTALL_OWNER) vtund $(SBIN_DIR)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(VAR_DIR)/run
+	$(INSTALL) -d -m 755 $(DESTDIR)$(STAT_DIR)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(LOCK_DIR)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(SBIN_DIR)
+	$(INSTALL) -m 755 vtund $(DESTDIR)$(SBIN_DIR)
 
 # DO NOT DELETE THIS LINE -- make depend depends on it.
--- vtun-2.4/configure.in	Thu Dec 28 06:35:10 2000
+++ vtun-2.4/configure.in-p	Tue Feb 13 20:52:44 2001
@@ -100,6 +100,22 @@
 dnl Check for setproctitle in libutil
 AC_SEARCH_LIBS(setproctitle, util bsd, AC_DEFINE(HAVE_SETPROC_TITLE) )
 
+if test "$SSL" = "yes"; then
+   AC_MSG_RESULT()
+   AC_CHECKING( for SSL Library and Header files ... )
+   AC_SEARCH_HEADERS(md5.h blowfish.h,
+	$SSL_HDR_DIR /usr/include/ssl /usr/include/openssl /usr/include /usr/local/include /usr/local/ssl/include,
+	AC_CHECK_LIB(crypto, BF_set_key,
+	    [
+	       LIBS="$LIBS -lcrypto"
+	       AC_DEFINE(HAVE_SSL)
+	    ],
+	    AC_MSG_ERROR( SSL library not found. )
+	),
+	AC_MSG_ERROR( SSL headers not found. )
+   )
+fi
+
 if test "$SHAPER" = "yes"; then
    AC_DEFINE(HAVE_SHAPER)
 fi
@@ -127,22 +143,6 @@
 	    AC_MSG_ERROR( LZO library not found. )
 	),
         AC_MSG_ERROR( LZO headers not found. )
-   )
-fi
-
-if test "$SSL" = "yes"; then
-   AC_MSG_RESULT()
-   AC_CHECKING( for SSL Library and Header files ... )
-   AC_SEARCH_HEADERS(md5.h blowfish.h,
-	$SSL_HDR_DIR /usr/include/ssl /usr/include/openssl /usr/include /usr/local/include /usr/local/ssl/include,
-	AC_CHECK_LIB(crypto, BF_set_key,
-	    [
-	       LIBS="$LIBS -lcrypto"
-	       AC_DEFINE(HAVE_SSL)
-	    ],
-	    AC_MSG_ERROR( SSL library not found. )
-	),
-	AC_MSG_ERROR( SSL headers not found. )
    )
 fi
 
